### ISSUE TITLE
Misleading diagnostic when using named arguments on function pointer call

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#48071]
**Version Used**: 6521491e7fbf9765797111faca482c9eb7f57116

**Steps to Reproduce**:

```cs
public class C {
    public unsafe void M(delegate*<string, int, void> ptr) {
        ptr("a", arg1: 1);
    }
}
```

[sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNGSAmIDUAfAAQGYACAgJhIGESBvAWACgSWzSBXAOyggDM4yAFhIBZABSY4AGzgBzCPABUAHgIBGAAxISAS04YhAPhIAHGAgCUdJq1unzYgEQRH2iAllqQJNRYDcNqwAvkxBQA)

**Expected Behavior**: An error is given indicating that function pointer invocations cannot have named arguments

**Actual Behavior**: `error CS8756: Function pointer 'delegate*<string, int, void>' does not take 2 arguments`

**Comment by Youssef1313** [comment]
I'll take this one.
